src/konnektor/network_planners/concatenator/cyclic_concatenator.py

```diff
@@ -1,37 +1,54 @@
 import logging
 from typing import Iterable
 
-
 from gufe import AtomMapper, AtomMappingScorer, LigandNetwork
+
 from ._abstract_network_concatenator import NetworkConcatenator
+from .max_concatenator import MaxConcatenate
 from .._networkx_implementations import MstNetworkGenerator
 
 log = logging.getLogger(__name__)
 
 
 # Todo: check this algorithm again
 
-class MstConcatenate(NetworkConcatenator):
-    def __init__(self, mapper: AtomMapper, scorer: AtomMappingScorer, n_connecting_edges: int = 3,
-                 node_present_in_cycles: int = 2, cycle_sizes: Union[int, List[int]] = 3, nprocesses: int = 1):
+class CyclicConcatenate(NetworkConcatenator):
+    def __init__(self, mapper: AtomMapper, scorer: AtomMappingScorer,
+                 n_connecting_cycles: int = 2,
+                 cycle_sizes: Union[int, List[int]] = 3, nprocesses: int = 1,
+                 _initial_edge_lister: NetworkConcatenator = None):
         """
-        This concatenator is connnecting two Networks with a kruskal like approach up to the number of connecting edges.
+        This concatenator is connnecting two Networks with a kruskal like
+        approach up to the number of connecting edges.
 
         Parameters
         ----------
         mapper: AtomMapper
-            the atom mapper is required, to define the connection between two ligands.
+            the atom mapper is required, to define the connection
+            between two ligands.
         scorer: AtomMappingScorer
-            scoring function evaluating an atom mapping, and giving a score between [0,1].
-        n_connecting_edges: int, optional
-            number of connecting edges. (default: 3)
+            scoring function evaluating an atom mapping, and giving a
+            score between [0,1].
+        n_connecting_cycles: int, optional
+            build at least n cycles between th networks. (default: 2)
+        cycle_sizes: Union[int, List[int]], optional
+            build cycles of given size. (default:3)
```

Comment:
document how this is different if it is a list of ints?

Reply:
ah good point!